pkg/providers/snow/snow.go

```diff
@@ -141,8 +107,17 @@ func (p *SnowProvider) GenerateCAPISpecForCreate(ctx context.Context, _ *types.C
 	return controlPlaneSpec, workersSpec, nil
 }
 
-func (p *SnowProvider) GenerateCAPISpecForUpgrade(ctx context.Context, bootstrapCluster, workloadCluster *types.Cluster, currrentSpec, newClusterSpec *cluster.Spec) (controlPlaneSpec, workersSpec []byte, err error) {
-	return nil, nil, nil
+func (p *SnowProvider) generateCAPISpec(ctx context.Context, clusterSpec *cluster.Spec) (controlPlaneSpec, workersSpec []byte, err error) {
+	kubeconfigClient := p.kubeUnAuthClient.KubeconfigClient(clusterSpec.ManagementCluster.KubeconfigFile)
```

Comment:
If we are receiving the `types.Cluster` separately, I would vote to get the kubeconfig from there. I think `clusterSpec.ManagementCluster` should eventually be removed.